src/TraceEvent/Parsers/ClrTraceEventParser.cs

```diff
@@ -9882,4 +9950,85 @@ public enum GCGlobalMechanisms
         Demotion = 0x8,
         CardBundles = 0x10,
     }
+
+    #region private types
+    /// <summary>
+    /// ClrTraceEventParserState holds all information that is shared among all events that is
+    /// needed to decode Clr events.   This class is registered with the source so that it will be
+    /// persisted.  Things in here include
+    /// 
+    ///     * TypeID to TypeName mapping, 
+    /// </summary>
+    // [SecuritySafeCritical]
+    internal class ClrTraceEventParserState : IFastSerializable
+    {
+        internal void SetTypeIDToName(int processID, Address typeId, long timeQPC, string typeName)
```

Comment:
The model is that any ID -> Name mapping has a lifetime (it begins and ends).   In this case we ended up not having a clean model, and I don't want to hide that fact.